File: src/main/java/de/mikeyllp/miniGamesV4/commands/ToggleInvitesCommand.java
package de.mikeyllp.miniGamesV4.commands;

import de.mikeyllp.miniGamesV4.database.Database;
import de.mikeyllp.miniGamesV4.storage.ToggleInvitesStorage;
import dev.jorel.commandapi.CommandAPICommand;

import static de.mikeyllp.miniGamesV4.utils.MessageUtils.sendNoPermissionMessage;

public class ToggleInvitesCommand extends CommandAPICommand {


    public ToggleInvitesCommand(String commandName, ToggleInvitesStorage storage) {
        super(commandName);
        //A Command witch toggle that he can get Invited
        executesPlayer(((sender, args) -> {
            //Checks if the player has permission to use this command
            if (!sender.hasPermission("minigamesv4.minigames")) {
                sendNoPermissionMessage(sender);
                return;
            }
            storage.addToggle(sender);
        }));
    }
}

File: src/main/java/de/mikeyllp/miniGamesV4/commands/ToggleInvitesCommand.kt
package de.mikeyllp.miniGamesV4.commands

import de.mikeyllp.miniGamesV4.permission.MinigamesPermissionRegistry
import de.mikeyllp.miniGamesV4.storage
import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.kotlindsl.playerExecutor
import dev.jorel.commandapi.kotlindsl.subcommand

fun CommandAPICommand.toggleInvitesCommand() = subcommand("toggle") {
    withPermission(MinigamesPermissionRegistry.COMMAND_TOGGLE)
    playerExecutor { player, args ->
        storage.addToggle(player)
    }
}
